ansible/roles/software/dashboard/dashboard-deploy/molecule/default/tests/test_default.py

```diff
@@ -6,7 +6,36 @@
     os.environ['MOLECULE_INVENTORY_FILE']).get_hosts('all')
 
 
-def test_hosts_file(host):
-    hosts = host.file('/etc/hosts')
+dashboard_user = 'ubuntu'
 
-    assert hosts.exists
+
+def test_log_dir(host):
+    log_dir = host.file('/var/log/dashboard')
+
+    assert log_dir.exists
+    assert log_dir.is_directory
+    assert log_dir.user == dashboard_user
+    assert log_dir.group == dashboard_user
+
+
+def test_supervisor_conf(host):
+    supervisor = host.file('/etc/supervisor/conf.d/dashboard')
+
+    assert supervisor.exists
+    assert supervisor.user == 'root'
+    assert supervisor.group == 'root'
+    assert supervisor.mode == 0o644
+    assert supervisor.contains('program:dashboard-cfo-act-download')
+    assert supervisor.contains('program:dashboard-cfo-act-full-scan')
+    assert supervisor.contains('user=%s' % dashboard_user)
+
+
+def test_cron(host):
+    cron = host.file('/etc/cron.d/dashboard')
+
+    assert cron.exists
+    assert cron.user == 'root'
+    assert cron.group == 'root'
+    assert cron.mode == 0o644
+    assert cron.contains('supervisorctl start dashboard-cfo-act-download')
+    assert cron.contains('supervisorctl start dashboard-cfo-act-full-scan')
```

Comment:
I'm a little confused here... So cron is still used to ensure the supervisor jobs are started? Isn't there a better mechanism for that, eg chkconfig? Or maybe the intent is just to recover in case one of them dies...?

Reply:
These jobs are not long-running processes, they are one-off tasks that exit 0 once the task is complete. cron schedules the job at a specific time and supervisor acts more like a task manager. Supervisor provides:
- task arguments "as code" since they're baked into the config
- parallelism of a task (or in this case, enforcing a singleton task)
- retries on failure (no retries in this case)
- logging

We use supervisor for both these one-off tasks and for long-running tasks/services.

Once we're completely on bionic, we'd probably [move to systemd](https://github.com/GSA/datagov-deploy/issues/805) which has more features like resource limits, watch dogs, and timers (dropping the need for cron).

Reply:
That makes sense to me... I was looking for a `chkconfig` or something somewhere.